### ISSUE TITLE
"Current licence does not include this feature" when trying to view differences

### ISSUE THREAD

**Issue by aurelin** [nathanwoulfe/Plumber-2#97]
I am using a test.lic test licence on a .local / .test domain.

When I click on the View Changes button to see the changes waiting for approval, I get an error saying that the current licence does not include that feature.

Stacktrace
`at Plumber.Core.Licensing.LicensedFeature() in D:\a\1\s\Plumber.Core\Licensing.cs:line 31
at Plumber.Web.Api.InstancesController.GetDiff(Guid guid) in D:\a\1\s\Plumber.Web\Api\InstancesController.cs:line 146`

I am using the Plumber-2 v1.6.7.824 (installed through Visual Studio NuGet package manager) for umbraco 8.13.

When I go to settings, the test licence is accepted and I'm able to use the Pro features there.

**Comment by nathanwoulfe** [comment]
Hi @aurelin is the test site running locally in debug mode?

As an aside, you might have issues with email notifications on Umbraco 8.13 as there were bugs in the localization logic, which were fixed in 8.15.

**Comment by aurelin** [comment]
Hi Nathan,I'm not near that computer and can't check, but iirc it should be in debug mode. Is that an issue?Also, thanks for the heads up! I did see an update notification so I'll make sure to apply that.On 30 Aug 2021 18:00, Nathan Woulfe ***@***.***> wrote:
Hi @aurelin is the test site running locally in debug mode?
As an aside, you might have issues with email notifications on Umbraco 8.13 as there were bugs in the localization logic, which were fixed in 8.15.

**Comment by nathanwoulfe** [comment]
Definitely should be in debug (both release configuration and debug=true in the web.config). 

I'll double check that this works with the compiled code, just occurred to me that I've been testing against source, so the compiled library may behave differently

**Comment by aurelin** [comment]
Just double-checked and yep it's in debug mode

**Comment by aurelin** [comment]
Hi Nathan,

I just wanted to check in if you saw my last comment - the release config and web.config are both set to debug mode.

Thank you!

**Comment by nathanwoulfe** [comment]
Yeah, still trying to debug this one - it's difficult given it works fine for me. You're not the only one seeing similar issue, so certainly something I need to sort out...

**Comment by nathanwoulfe** [comment]
@aurelin I'm looking at this now, I can't reproduce the issue:
- running Umbraco 8.13.0
- plumber 1.6.8 nightly build (beta builds in release configuration)
- visual studio 16.10.4
- debug, any cpu
- web.config compilation debug="true"
- site served from https://localhost:44362
- backoffice shows license impersonation banner
- compare diff returns the expected data, no error

Are you running from visual studio or is the site deployed to a test server? Needs to be running from the IDE.

Screen dump, for what it's worth:
![image](https://user-images.githubusercontent.com/3248070/132806026-df11f0ec-fb8d-4683-919d-705bbf949924.png)

**Comment by aurelin** [comment]
I'm running it as a virtual application in IIS on my local machine under https://umbraco.local. In the backoffice, it recognises the test licence, and other parts (such as document type approval flows and exclude nodes) work well.

**Comment by aurelin** [comment]
I have the same issue when running from IIS express (localhost:[port])

That being said, I just realised the umbraco version I stated is wrong (I forgot I updated). I'm at 8.16.

**Comment by nathanwoulfe** [comment]
Wondering if it's due to virtualising - the test license is really intended for use directly from visual studio (ie new solution, install umbraco, install plumber, add license, F5).

I'm going to add some additional logging to the next release to make these sorts of issues easier to debug

**Comment by aurelin** [comment]
I'm not sure.. I get the same error when I run out of IIS express (installed umbraco within an existing solution as a separate project, installed plumber through NuGet, added licence, F5). The entire solution is in debug mode on localhost.

**Comment by nathanwoulfe** [comment]
Strange... The 1.6.8 beta has some additional logging to check the domains being compared when validating the test license, maybe bump to that version and share the logs?

I'll add some more, and push a new beta this morning.

**Comment by aurelin** [comment]
Installed the beta, then re-tried in IIS express. That now works, not sure why it didn't before. 

When trying to access through a .local domain, it still fails.
From the log file:

`{"@t":"2021-09-15T23:10:54.7750323Z","@mt":"Error getting changes for workflow instance 658e774c-783c-4593-8f1a-7f8172e076ea: {message}","@l":"Error","@x":"Plumber.Core.InvalidLicenseException: Current license does not include this feature\r\n   at Plumber.Core.Licensing.LicensedFeature() in D:\\a\\1\\s\\Plumber.Core\\Licensing.cs:line 36\r\n   at Plumber.Core.Services.Implement.InstancesService.GetDiff(Guid guid) in D:\\a\\1\\s\\Plumber.Core\\Services\\Implement\\InstancesService.cs:line 335\r\n   at Plumber.Web.Api.InstancesController.GetDiff(Guid guid) in D:\\a\\1\\s\\Plumber.Web\\Api\\InstancesController.cs:line 144","message":"Current license does not include this feature","SourceContext":"Plumber.Web.Api.InstancesController","ProcessId":7124,"ProcessName":"w3wp","ThreadId":11,"AppDomainId":2,"AppDomainAppId":"LMW3SVC2ROOT","MachineName":"MACHINEHOSTNAME","Log4NetLevel":"ERROR","HttpRequestNumber":5,"HttpRequestId":"66c41792-9cb7-4a7b-a801-a2ce49e1367b"}`

I guess I can just run it from IIS express for now, though I'm hoping to be able to use a .local domain for my purposes, IIS express has its own issues for my project.

**Comment by aurelin** [comment]
Hi Nathan,

I upgraded to umbraco 8.17 / workflow 2.0.2 and am still getting this error message when using a .test or .local domain, except now it appears as a nice, red error message, not an uncaught exception.

**Comment by nathanwoulfe** [comment]
How is your site set up? Needs to running in debug, from the IDE. The domain no longer matters, so long as these conditions are met. 

Does your backoffice show the license impersonation banner? What do you see in the licensing dashboard in the workflow section?

**Comment by aurelin** [comment]
I am not currently able to get IISExpress to run with a .local domain. While my settings are so that it builds in Debug mode, it isn't running from VS directly (it's set up as a site in IIS). 
It does not show the banner there, although it does say a test licence is active in settings.

When I run it from VS, it works, but it isn't under my local domain. I guess I can switch back and forth depending on what I'm testing.
